inc/phylogeny/tree.cuh

```diff
@@ -0,0 +1,76 @@
+/**
+ * Multiple Sequence Alignment phylogeny tree header file.
+ * @author Rodrigo Siqueira <rodriados@gmail.com>
+ * @copyright 2019 Rodrigo Siqueira
+ */
+#pragma once
+
+#ifndef PG_TREE_CUH_INCLUDED
+#define PG_TREE_CUH_INCLUDED
+
+#include "buffer.hpp"
+#include "pairwise.cuh"
+
+namespace phylogeny
+{
+    /*
+     * Defines the value of an undefined node reference.
+     */
+    enum : uint16_t { undefined = 0xffff };
+
+    /**
+     * Represents a tree node, containing a pair of child OTU references.
+     * @since 0.1.1
+     */
+    struct Node
+    {
+        Pair child {{undefined, undefined}};    /// The nodes that lead to the current one.
```

Comment:
Maybe the usage of `Pair` here is not consistent. It would better be an array instead of `Pair`.